harvester/harvester
# Use Ingress to expose Harvester and support Certificates configuration

https://github.com/harvester/harvester/issues/761
https://github.com/harvester/harvester/issues/1475

Depend on PR https://github.com/harvester/harvester-installer/pull/173

1. Use a secret controller to ensure the VIP is added to TLS SAN.
2. Expose Harvester service using ingress. Set the default SSL certificate of ingress controller so that domain name is not required.
3. On certificate setting changes, update ingress controller configurations.

Test plan:
1. Generate self-signed TLS certificates(for example use openssl or https://www.selfsignedcertificate.com/). Configure the certificates in Harvester settings. Verify the Harvester serving certificate is updated accordingly.
2. Verify users can update the certificate by changing the setting.
3. Verify users can revert to the Harvester-signed certificate by cleaning up the setting.
4. Verify the generated kubeconfig works with the certificate change.
